tests/test_all_rules.py

```diff
@@ -440,19 +443,30 @@ def test_integration(self):
         """Test that rules in integrations folders have matching integration defined."""
         failures = []
 
+        packages_manifest = load_integrations_manifests()
+        epr_search_url = "https://epr.elastic.co/search"
+        epr_search_response = requests.get(epr_search_url, params={"all": "true"})
```

Comment:
can we avoid network calls in these tests. Couldn't this also be achieved by loading the latest manifest dump and parsing from there

Reply:
We would need to update the integrations manifest dump to include all integrations. The network call is meant to pull all integration manifests, while the integrations manifest dump only includes any rules with an integration tag.

An example on the resiliency is this:

Our current integrations manifest is built based on all `integration` tags in existing rules. Therefore we have `azure`, `gcp`, `aws`. User A writes a rule for K8s and adds the `integrations` tag, but sets the value to `kubernetes`. Since our unit test only checks the local integrations manifest dump, K8s would be missing and this would fail. The network call is grabbing all manifests regardless of integration, so it passes because `kubernetes` is still a valid integration, but Detection Rules has never had a rule for it before.

Option A - We could reference the local integrations manifest dump as proposed and if failing, we know this is a new rule for an integration we have never had rules for before. The action is to then update the integrations manifest with the new rule locally by running the update command.

Option B - We build our integrations manifest dump with all integrations and then reference that which will reduce the possibility for this test to fail.

Option C - We check local integrations manifest dump, if not found, we check EPR. This still does a single network request as already implemented though.

Reply:
I think we should strive to avoid network calls from tests wherever possible.

>Option C - We check local integrations manifest dump, if not found, we check EPR. This still does a single network request as already implemented though.

Let's do a variation of this, where if it is not found in the manifest dump, the specific error says something like

```
unknown integration - check spelling or refresh manifest file ...
```

This is a better approach IMO since it forces us to keep the dump current (and avoids the network call!)

Reply:
@brokensound77 adjusted this to suggest updating the manifest file. If this is a member of TRADE, that is fine but we should take extra scrutiny if a community member wants to do so. If that is the case we should update it ourselves.

<img width="1107" alt="Screen Shot 2022-09-26 at 10 37 26 PM" src="https://user-images.githubusercontent.com/99630311/192419035-947abdbc-ba8c-478d-8675-5cb6a4c4feaa.png">